src/main/java/com/github/checkstyle/regression/git/DiffParser.java

```diff
@@ -0,0 +1,106 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.checkstyle.regression.git;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.lib.ObjectReader;
+import org.eclipse.jgit.lib.Ref;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.revwalk.RevTree;
+import org.eclipse.jgit.revwalk.RevWalk;
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder;
+import org.eclipse.jgit.treewalk.AbstractTreeIterator;
+import org.eclipse.jgit.treewalk.CanonicalTreeParser;
+
+/**
+ * Git diff parser.
+ * @author LuoLiangchen
+ */
+public class DiffParser {
+    /** The prefix of branch head commit ref. */
+    private static final String COMMIT_REF_PREFIX = "refs/heads/";
+
+    /** The path of checkstyle repository. */
+    private final String repositoryPath;
+
+    /**
+     * Creates a new {@code DiffParser} instance.
+     * @param repositoryPath the path of checkstyle repository
+     */
+    public DiffParser(String repositoryPath) {
+        this.repositoryPath = repositoryPath;
+    }
+
+    /**
+     * Parses the diff between a given branch and the master.
+     * @param branchName the name of the branch to be compared with master
+     * @return a list of {@code GitChange} to represent the changes
+     * @throws IOException JGit library exception
+     * @throws GitAPIException JGit library exception
+     */
+    public List<GitChange> parse(String branchName) throws IOException, GitAPIException {
```

Comment:
I just designed it by my intuition previously. But actually my point is on the opposite, I think if we need work with more than one repository, then static method is better. 
I was thinking that we may work with only one repository path and more than one branch, for there is only one checkstyle project but many feature branches. Then we could just set the repo path once in the constructor and run `parse` method with only the branch parameter.
But I guess in most case we would work with only one repo path and one branch. So if you think static method is better, I am ok with that.